### PR TITLE
[v4] Support arbitrary gradient angles

### PR DESCRIPTION
**Pull request by thecrypticace**
This makes things like `bg-gradient-[125deg]` and `bg-gradient-[1.3rad]` work. Now the output will be:

```css
.bg-gradient-\[1\.3rad\] {
  background-image: linear-gradient(1.3rad, var(--tw-gradient-stops, ));
}

.bg-gradient-\[125deg\] {
  background-image: linear-gradient(125deg, var(--tw-gradient-stops, ));
}
```

### CODE REVIEW

**Review by adamwathan — COMMENTED** [review]
Wonder if we should do this under a `bg-linear` namespace instead of `bg-gradient` but keep gradient for BC? That way we could also do `bg-conic` and `bg-radial` as well 🤔

**Comment by thecrypticace** [comment]
You know… I'm totally on board with that.

**Review by RobinMalfait — COMMENTED** [review]
Nice!

2 questions:

1. For the `bg-radial` examples, will we do that in a new PR or in the same PR?
2. This currently still allows you to type `bg-gradient-[123deg]`. Should we infer the `angle` data type in the `'bg'` plugin and just return early so we don't fallback to `background-color`?

Something like:
```diff
diff --git a/packages/tailwindcss/src/utilities.ts b/packages/tailwindcss/src/utilities.ts
index 78d653b9d..53ac9a584 100644
--- a/packages/tailwindcss/src/utilities.ts
+++ b/packages/tailwindcss/src/utilities.ts
@@ -2532,9 +2532,13 @@ export function createUtilities(theme: Theme) {
           'bg-size',
           'length',
           'url',
+          'angle',
         ])
 
       switch (type) {
+        // Explicitly unsupported
+        case 'angle':
+          return
         case 'percentage':
         case 'position': {
           return [decl('background-position', value)]
```

**Comment by RobinMalfait** [comment]
Just for completeness so we don't forget:

The last one I mentioned, where we infer the `angle` and mark it as not supported is a band-aid solution. Instead, we should bail earlier because we already know that it was invalid. `bg-foo-bar-baz-[123deg]` is not valid if `bg-foo-bar-baz` doesn't exist as a plugin root.

File: packages/tailwindcss/src/utilities.test.ts [hunk 1]
```diff
@@ -7974,6 +8030,10 @@ test('bg', () => {
       // background-image
       '-bg-none',
       '-bg-gradient-to-br',
+      '-bg-linear-to-br',
+      '-bg-linear-[125deg]', // This could be made valid maybe
+      '-bg-linear-[1.3rad]', // This could be made valid maybe
```
Comment (adamwathan) [inline packages/tailwindcss/src/utilities.test.ts:8035]: I think we should support this personally 👍

File: packages/tailwindcss/src/utilities.test.ts [hunk 2]
```diff
@@ -7690,6 +7690,8 @@ test('bg', () => {
 
         // background-image
         'bg-none',
+
+        // Legacy linear gradient version
```
Comment (adamwathan) [inline packages/tailwindcss/src/utilities.test.ts:7694]: I'd say "Legacy linear-gradient API" here, with hyphen.

File: packages/tailwindcss/src/utilities.test.ts [hunk 3]
```diff
@@ -7698,10 +7700,24 @@ test('bg', () => {
         'bg-gradient-to-bl',
         'bg-gradient-to-l',
         'bg-gradient-to-tl',
+
+        // Modern linear-gradient version
```
Comment (adamwathan) [inline packages/tailwindcss/src/utilities.test.ts:7704]: Swap version to API here too IMO.

File: CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Use `length` data type for `background-size` instead of `background-position` ([#13771](https://github.com/tailwindlabs/tailwindcss/pull/13771))
 - Support negative values for `{col,row}-{start,end}` utilities ([#13780](https://github.com/tailwindlabs/tailwindcss/pull/13780))
 - Use long form `<length> | <percentage>` syntax for properties ([#13660](https://github.com/tailwindlabs/tailwindcss/pull/13660))
+- Support arbitrary gradient angles ([#13783](https://github.com/tailwindlabs/tailwindcss/pull/13783))
```
Comment (adamwathan) [inline CHANGELOG.md:15]: Should also note the updated API here 👍

File: packages/tailwindcss/src/utilities.test.ts (hunk 1 above)
Comment (thecrypticace) [reply packages/tailwindcss/src/utilities.test.ts]: Updated

File: CHANGELOG.md (hunk 4 above)
Comment (thecrypticace) [reply CHANGELOG.md]: Updated

File: CHANGELOG.md [hunk 5]
```diff
@@ -12,10 +12,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Use `length` data type for `background-size` instead of `background-position` ([#13771](https://github.com/tailwindlabs/tailwindcss/pull/13771))
 - Support negative values for `{col,row}-{start,end}` utilities ([#13780](https://github.com/tailwindlabs/tailwindcss/pull/13780))
 - Use long form `<length> | <percentage>` syntax for properties ([#13660](https://github.com/tailwindlabs/tailwindcss/pull/13660))
+- Support arbitrary linear gradient angles (e.g. `bg-linear-[125deg]`) ([#13783](https://github.com/tailwindlabs/tailwindcss/pull/13783))
```
Comment (adamwathan) [inline CHANGELOG.md:15]: This can go in "Added" I think IMO 👍

**Review by adamwathan — APPROVED** [review]
Just move that one changelog thing but otherwise rock and roll 🤘

**Pull request merged**